gokart/gcs_obj_metadata_client.py

```diff
@@ -101,23 +106,51 @@ def _get_patched_obj_metadata(
         # However, users who utilize custom_labels are no longer expected to search using the labels generated from task parameters.
         # Instead, users are expected to search using the labels they provided.
         # Therefore, in the event of a key conflict, the value registered by the user-provided labels will take precedence.
-        _merged_labels = GCSObjectMetadataClient._merge_custom_labels_and_task_params_labels(normalized_task_params_labels, normalized_custom_labels)
+        normalized_labels = (
+            [normalized_custom_labels, normalized_task_params_labels]
+            if not required_task_outputs
+            else [
+                normalized_custom_labels,
+                normalized_task_params_labels,
+                {'__required_task_outputs': json.dumps(GCSObjectMetadataClient._get_serialized_string(required_task_outputs))},
+            ]
+        )
+        _merged_labels = GCSObjectMetadataClient._merge_custom_labels_and_task_params_labels(normalized_labels)
         return GCSObjectMetadataClient._adjust_gcs_metadata_limit_size(dict(metadata) | _merged_labels)
 
+    @staticmethod
+    def _get_serialized_string(required_task_outputs: FlattenableItems[RequiredTaskOutput]) -> FlattenableItems[str]:
+        def _list_flatten(nested_list: list):
+            flattened_list = []
+            for item in nested_list:
+                if isinstance(item, list):
+                    flattened_list.extend(_list_flatten(item))
+                else:
+                    flattened_list.append(item)
+            return flattened_list
+
+        if isinstance(required_task_outputs, dict):
+            return {k: GCSObjectMetadataClient._get_serialized_string(v) for k, v in required_task_outputs.items()}
+        if isinstance(required_task_outputs, tuple):
+            return tuple(required_task_output.serialize() for required_task_output in required_task_outputs)
+        if isinstance(required_task_outputs, Iterable):
+            return _list_flatten([GCSObjectMetadataClient._get_serialized_string(ro) for ro in required_task_outputs])
```

Comment:
tuple is one type of Iterable, and python has many iterable type, e.x) list, set, tuple, so on. You can write better like the following.

```suggestion
        if isinstance(required_task_outputs, Iterable):
            iter_type = type(required_task_outputs)
            return iter_type(_list_flatten([GCSObjectMetadataClient._get_serialized_string(ro) for ro in required_task_outputs]))
```

Reply:
Sorry for the late reply.

Yes, you are correct, and you can change iterable to list,

And, What I actually meant was regarding the utils.map_flattenable_items method. I'll make another comment.

Reply:
Thank you for replying.
I deal with your suggested changes.